instantsearch/src/main/java/com/algolia/instantsearch/helpers/InstantSearch.java

```diff
@@ -85,40 +86,96 @@ public InstantSearch(@NonNull final Activity activity, @NonNull final Searcher s
 
         registerSearchView(activity, menu, menuItemId);
         processActivity(activity);
     }
 
+    /**
+     * Constructs the helper, then link it to the given Activity and register its fragment's Widgets.
+     *
+     * @param activity a searchable Activity.
+     * @param fragment a Fragment containing at least one {@link AlgoliaResultsListener} to update with incoming results.
+     * @param searcher the Searcher to use with this activity.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
+    public InstantSearch(@NonNull final Activity activity, @NonNull final Searcher searcher, @NonNull Fragment fragment) {
+        this(searcher);
+
+        registerSearchView(activity);
+        processAllListeners(fragment.getView());
+    }
+
+    /**
+     * Constructs the helper, then link it to the given Activity and register its fragment's Widgets.
+     *
+     * @param activity a searchable Activity.
+     * @param fragment a Fragment containing at least one {@link AlgoliaResultsListener} to update with incoming results.
+     * @param searcher the Searcher to use with this activity.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
+    public InstantSearch(@NonNull final Activity activity, @NonNull final Searcher searcher, @NonNull android.support.v4.app.Fragment fragment) {
+        this(searcher);
+
+        registerSearchView(activity);
+        processAllListeners(fragment.getView());
+    }
+
+
+    /**
+     * Constructs the helper, then link it to the given Widget.
+     *
+     * @param widget   a Widget to register as listener.
+     * @param searcher the Searcher to use with this InstantSearch.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
     public InstantSearch(@NonNull final View widget, @NonNull final Searcher searcher) {
         this(searcher);
         registerWidget(widget);
     }
 
-    private InstantSearch(@NonNull final Searcher searcher) {
+    /**
+     * Constructs the helper.
+     *
+     * @param searcher the Searcher to use with this InstantSearch.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
+    public InstantSearch(@NonNull final Searcher searcher) {
         this.searcher = searcher;
         enableProgressBar();
     }
 
     /**
-     * Triggers a new search with the {@link #searchView}'s text.
+     * Triggers a new search with the {@link #searchBoxViewModel SearchView}'s text.
      */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
     public void search() {
         searcher.search();
     }
 
     /**
-     * Registers the Search Widget of an Activity's Menu to trigger search requests on text change.
+     * Triggers a new search with the given text.
+     *
+     * @param query the text to search for.
+     */
+    public void search(String query) {
+        searcher.setQuery(searcher.getQuery().setQuery(query)).search();
```

Comment:
A lot of things are happening here, If you can write separate statements for that it would be great!